compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java

```diff
@@ -939,6 +940,25 @@ public static BuildJson readBuildJson(Path buildJsonPath) throws JsonSyntaxExcep
         }
     }
 
+    public static boolean isProjectUpdated(Project project) {
+        if (project.sourceRoot().resolve(TARGET_DIR_NAME).resolve(BUILD_FILE).toFile().exists()) {
+            try {
+                BuildJson buildJson = readBuildJson(project.sourceRoot().resolve(TARGET_DIR_NAME).resolve(BUILD_FILE));
+                long lastProjectUpdatedTime = FileUtils.lastModifiedTimeOfBalProject(project.sourceRoot());
+                if (buildJson != null
+                        && buildJson.getLastModifiedTime() != null
+                        && !buildJson.getLastModifiedTime().entrySet().isEmpty()) {
+                    long defaultModuleLastModifiedTime = buildJson.getLastModifiedTime()
+                            .get(project.currentPackage().packageName().value());
+                    return lastProjectUpdatedTime > defaultModuleLastModifiedTime;
+                }
+            } catch (IOException e) {
+                throw new ProjectException("Reading 'build' file failed: " + e.getMessage());
```

Comment:
Let's delete the build file and return `true` in case reading build file fails

Reply:
Changed it as follows:
```java
} catch (IOException e) {
    // if reading `build` file fails
    // delete `build` file and return true
    try {
        Files.deleteIfExists(buildFile);
     } catch (IOException ex) {
         // ignore
     }
     return true;
}
```